src/charactersheet/services/common/account/party_service.js

```diff
@@ -0,0 +1,124 @@
+import {
+    CoreManager,
+    Notifications
+} from 'charactersheet/utilities';
+import { Party } from 'charactersheet/models/common';
+import autoBind from 'auto-bind';
+
+
+class _PartyService {
+
+    constructor({ REFRESH_INTERVAL }) {
+        autoBind(this);
+
+        this.party = null;
+        this.refreshInterval = REFRESH_INTERVAL;
+
+        this.setupNotifications();
+    }
+
+    setupNotifications() {
+        Notifications.coreManager.changed.add(
+            this.checkForPartyAndSetupPollingIfNeccessary
+        );
+    }
+
+    async checkForPartyAndSetupPollingIfNeccessary() {
+        await this.refresh();
+        if (!!this.party) {
+            console.log('Party auto-refresh enabled.')
+            this.configureTimers();
+        }
+    }
+
+    // Timers
+
+    configureTimers() {
+        this.resetTimers();
+        this.timer = setInterval(this.refresh, this.refreshInterval);
+    }
+
+    resetTimers() {
+        if (!!this.timer) {
+            clearInterval(this.timer);
+        }
+    }
+
+    // Poll
+
+    async refresh() {
+        let uuid;
+        try {
+            uuid = CoreManager.activeCore().uuid();
+        } catch(error) {
+            console.warn('No core to begin fetching details for.', error);
+            return;
+        }
+
+        try {
+            const { data: party } = await Party.ps.read(
```

Comment:
is this `read` different than other reads? what are the other bools?

Reply:
They bypass the Hypnos cache. Without bypassing it, Hypnos would prevent re-fetching the same party object if the fetch has already been made with no dependent changes.

```
    read = async (params={}, raw=false, useCache=true, flushDepsCache=false) => { ... }
```

[Hypnos &#8594;](https://adventurerscodex.github.io/hypnos/file/src/tokens.js.html#lineNumber224)

Reply:
might be worth adding a comment about the bypass